devito/equation.py

```diff
@@ -50,15 +57,33 @@ class Eq(sympy.Eq):
     def __new__(cls, *args, **kwargs):
         kwargs['evaluate'] = False
         subdomain = kwargs.pop('subdomain', None)
+        coefficients = kwargs.pop('coefficients', None)
         obj = sympy.Eq.__new__(cls, *args, **kwargs)
         obj._subdomain = subdomain
+        obj._coefficients = coefficients
+        functions = retrieve_functions(obj)
+        functions = filter_ordered(functions, key=lambda i: i.name)
```

Comment:
I think you can drop this line. By default, ``retrieve_functions`` returns unique items, though you can change it, see [here](https://github.com/opesci/devito/blob/master/devito/symbolics/search.py#L129)

Reply:
Ta, dropped.